apps/client/src/components/Layout/layout.tsx

```diff
@@ -1,15 +1,58 @@
-import { FC } from "react";
-import { LayoutType } from "./type";
-import { Header } from "../Header";
-import { Footer } from "../Footer"
+import { FC, useEffect, useState } from 'react';
+import { useLocation, useNavigate } from 'react-router-dom';
+import { LayoutType } from './type';
+import { Header } from '../Header';
+import { Footer } from '../Footer';
+import { Loader } from '../Loader';
+import RoomService from '../../services/room';
 
 export const Layout: FC<LayoutType> = (props) => {
-    const { children } = props;
-    return (
-        <div>
-            <Header />
-            { children }
-            <Footer />
-        </div>
-    )
-}
\ No newline at end of file
+  const { children } = props;
+  const [loading, setLoading] = useState(false);
+  const location = useLocation();
+  const navigate = useNavigate();
+  const roomKey = new URLSearchParams(location.search).get('roomKey') || '';
```

Comment:
import { useMemo } from 'react';
import {
    useLocation
} from "react-router-dom";

function useQuery() {
    const { search } = useLocation();

    return useMemo(() => new URLSearchParams(search), [search]);
}

export default useQuery;

then we can use 
const query = useQuery();
const roomKey = query.get('roomKey')

Just recommendation no force on this

Reply:
Yeah I am planning to create a hook for this which will return something like below. 
if some new key introduced we can update this type and hook

```
type QueryParamsType = {
  roomKey: string | null
}

```
```
export const useQueryParams = (): QueryParamsType => {
    const params: IPVQueryParams = {};
    const [searchParams, _setSearchParams] = useSearchParams(); // this is coming from react-router
    params.roomKey = searchParams.get("mapId");
    return params;
};
```